checks/apps/gromacs/gromacs_benchpep-h.py

```diff
@@ -0,0 +1,162 @@
+        self.descr = f'GROMACS {self.__bench} benchmark (NB: {self.nb_impl})'
+        #self.prerun_cmds = [
+        #    f'curl -LJO https://www.mpinat.mpg.de/{self.__bench} && unzip {self.__bench}.zip && ln -s {self.__bench}.tpr benchmark.tpr'
+        #]
+        bench_file_path = os.path.join(self.current_system.resourcesdir, 'gromacs-benchmarks', 'www.mpinat.mpg.de', self.__bench, f'{self.__bench}.tpr') 
+        self.prerun_cmds = [
+            f'ln -s {bench_file_path} benchmark.tpr'
+        ]
+        npme_ranks = 2*self.num_nodes
+        self.executable_opts += [
+            '-nsteps 5000',
+            '-nstlist 400',
+            '-noconfout',
+            '-nb', self.nb_impl,
+            f'-npme {npme_ranks}',
+            '-pme', self.nb_impl, 
+            '-update', self.nb_impl, 
+            '-bonded', self.nb_impl, 
+            '-pin on',
+            '-s benchmark.tpr'
+        ]
+        self.variables = {
+            'MPICH_GPU_SUPPORT_ENABLED': '1',
+            'GMX_ENABLE_DIRECT_GPU_COMM': '1',
+            'GMX_FORCE_GPU_AWARE_MPI': '1',
+            'OMP_NUM_THREADS': '7',
+            'GMX_GPU_PME_DECOMPOSITION': '1',
+            'GMX_PMEONEDD': '1'
+        }
+
+    @run_before('run')
+    def set_cpu_mask(self):
+        cpu_bind_mask = '0xfe000000000000,0xfe00000000000000,0xfe0000,0xfe000000,0xfe,0xfe00,0xfe00000000,0xfe0000000000'
+        self.job.launcher.options = [f'--cpu-bind=mask_cpu:{cpu_bind_mask}']
+         
+    @run_after('init')
+    def add_select_gpu_wrapper(self):
+        self.prerun_cmds += [
+            'cat << EOF > select_gpu',
+            '#!/bin/bash',
+            'export ROCR_VISIBLE_DEVICES=\$SLURM_LOCALID',
+            'exec \$*',
+            'EOF',
+            'chmod +x ./select_gpu'
+        ]
+        self.executable = './select_gpu ' + self.executable
+
+
+    @run_after('init')
+    def setup_nb(self):
+        valid_systems = {
+            'gpu': {
+                1: ['lumi:gpu'],
+                2: ['lumi:gpu'],
+            }
+        }
+        try:
+            self.valid_systems = valid_systems[self.nb_impl][self.num_nodes]
+        except KeyError:
+            self.valid_systems = []
+
+
+    @performance_function('ns/day')
+    def perf(self):
+        return sn.extractsingle(r'Performance:\s+(?P<perf>\S+)',
+                                'md.log', 'perf', float)
+
+    @deferrable
+    def energy(self):
+        return sn.extractsingle(r'\s+Potential\s+Kinetic En\.\s+Total Energy'
+                                r'\s+Conserved En\.\s+Temperature\n'
+                                r'(\s+\S+){2}\s+(?P<energy>\S+)(\s+\S+){2}\n'
+                                r'\s+Pressure \(bar\)\s+Constr\. rmsd',
+                                'md.log', 'energy', float, item=-1)
+
+    @sanity_function
+    def assert_energy_readout(self):
+        #energy_diff = sn.abs(energy - self.energy_ref)
+        return sn.assert_found('Finished mdrun', 'md.log')
+
```

Comment:
`grep 'Conserved energy' md.log | awk '{print $4}'` is the quantity that I suggest also checking. For this to be reliable we may need to increase the number of iterations slightly. To generate a reference value, please run a slightly longer run, e.g. 10k step (more might be needed for smaller inputs since this quantity is an average drift per atom)